custom_components/tuya_local/devices/forceclima_soundless.yaml

```diff
@@ -0,0 +1,86 @@
+name: ForceClima Soundless
+products:
+  - id: bf122f759ba4882810sdwc
+    name: Portable Air Conditioning
+primary_entity:
+  entity: climate
+  dps:
+    - id: 1
+      type: boolean
+      name: hvac_mode
+      mapping:
+        - dps_val: false
+          value: "off"
+        - dps_val: true
+          constraint: mode
+          conditions:
+            - dps_val: cold
+              value: cool
+            - dps_val: wind
+              value: fan_only
+            - dps_val: wet
+              value: dry
+            - dps_val: 'hot'
+              value: heat
+
+    - id: 2
+      type: integer
+      name: temperature
+      range:
+        min: 16
+        max: 30
+      unit: C
+
+    - id: 3
+      type: integer
+      name: current_temperature
+    - id: 4
+      type: string
+      name: mode
+      hidden: true
+    - id: 5
+      type: string
+      name: fan_mode
+      mapping:
+        - dps_val: low
+          value: low
+        - dps_val: med
+          value: medium
+        - dps_val: high
+          value: high
+    - id: 20
+      type: integer
+      name: fault_code
+
+    - id: 107
+      type: integer
+      name: temp_set_f
+    - id: 109
+      type: integer
+      name: unknown_107
+
+    - id: 110
+      name: swing_mode
+      type: boolean
+      mapping:
+        - dps_val: false
+          value: "off"
+        - dps_val: true
+          value: vertical
+
+secondary_entities:
+  - entity: switch
+    name: Sleep
+    category: config
+    icon: "mdi:power-sleep"
+    dps:
+      - id: 103
+        name: switch
+        type: boolean
+  - entity: binary_sensor
+    category: diagnostic
+    class: problem
+    dps:
+      - id: 20
+        type: integer
```

Comment:
Probably this is a bitfield.  It doesn't make much difference until you start trying to decode individual bits, but I like to use the correct types for future reference.

The values should be mapped to booleans for a binary_sensor
```
        mapping:
          - dps_val: 0
            value: false
          - value: true
``` 

As written above, adding the raw fault_code as a second attribute makes it easy to find.